vllm/sequence.py

```diff
@@ -1345,10 +1345,8 @@ def prune(self,
                 index = [self._seq_ids.index(seq_id) for seq_id in seq_ids]
             else:
                 # This path is added for use_padding_aware_scheduling
-                index = [
-                    self._seq_ids.index(seq_id)
-                    if seq_id in self._seq_ids else 0 for seq_id in seq_ids
-                ]
+                index = [self._seq_ids.index(seq_id) for seq_id in seq_ids if seq_id in self._seq_ids]
+                index = index + ([-1] * (len(seq_ids) - len(index))) 
```

Comment:
As you mentioned, because the line 1341, 1348-1349 is dead codes, I think you can update as below:

```
is_paddingware_scheduling = not (len(seq_ids) < len(self._seq_ids))
seq_ids = [seq_id for seq_id in seq_ids if seq_id in self._seq_ids]
index = [self._seq_ids.index(seq_id) for seq_id in seq_ids]
if is_paddingware_scheduling:
    index = index + ([-1] * (len(seq_ids) - len(index)))

... original codes
```